components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -4087,6 +4177,25 @@ public Map<API, List<APIProductResource>> addAPIProductWithoutPublishingToGatewa
         return apiToProductResourceMapping;
     }
 
+    private static void validateAPIProductContextTemplate(APIProduct product) throws APIManagementException {
+        String contextTemplate = product.getContextTemplate();
+
+        //Validate if the API Product has an unsupported context before executing the query
+        String invalidContext = "/" + APIConstants.VERSION_PLACEHOLDER;
+        if (invalidContext.equals(contextTemplate)) {
+            throw new APIManagementException(
+                    "Cannot add API Product : " + product.getId() + " with unsupported context : "
+                            + contextTemplate);
+        }
+
+        //If the context template ends with {version} this means that the version will be at the end of the context.
+        if (contextTemplate.endsWith("/" + APIConstants.VERSION_PLACEHOLDER)) {
```

Comment:
This needs to be moved to `addAPIProductWithoutPublishingToGateway` method to preserve the same behaviour as APIs

Reply:
done with https://github.com/wso2/carbon-apimgt/pull/12168/commits/27e7b0edcdab808f7a4efe9b2a366368cfca0652